src/sinks/statsd.rs

```diff
@@ -186,10 +187,10 @@ fn encode_event(event: Event, namespace: &str) -> Option<Vec<u8>> {
             }
         }
         _ => {
-            warn!(
-                "invalid metric sent to statsd sink ({:?}) ({:?})",
-                metric.kind, metric.value
-            );
+            emit!(StatsdInvalidMetric {
+                value: &metric.value,
+                kind: &metric.kind,
+            });
```

Comment:
We should also return `None` here. I wonder if a `None` would stop the sink?

Reply:
I tested this out and threw a load of events at it, and the Sink was not stopped when this returned None, any ensuing events were sent out just fine. I'm guessing the `flat_map` prevents any `None`, if it was multiple events we were encoding here, a None might prevent the rest in the batch from being sent - but in this case it is just one event at a time.